src/App.tsx

```diff
@@ -15,71 +16,87 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              aria-label="clear"
+              onClick={() => {
+                this.setState({ selectedGood: '' });
+              }}
+            />
+          </h1>
+        ) : (
+          <h1 className="title">No goods selected</h1>
+        )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                className={classNames(
+                  '',
+                  {
+                    'has-background-success-light': selectedGood === good,
+                  },
+                )}
+              >
+                <td>
+                  {selectedGood === good ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => {
+                        this.setState({
+                          selectedGood: '',
+                        });
+                      }}
```

Comment:
you can create a separate component for button
```suggestion
                    <Button
                      className="is-info"
                      onClick={handleReset};
                    >
                      
```

Reply:
Thanks for review. About separate components the task says not to do this
![image](https://user-images.githubusercontent.com/98056446/227497161-d98cf69e-e7b8-4504-ad92-1fedc594724b.png)